src/org/gk/render/RenderUtility.java

```diff
@@ -196,7 +196,10 @@ public static void drawName(Node node, Graphics2D g2) {
             g2.setPaint(DefaultRenderConstants.SELECTION_WIDGET_COLOR);
         }
         else if (node.getForegroundColor() == null)
-            g2.setPaint(DefaultRenderConstants.DEFAULT_FOREGROUND);
+            if (node.getIsForDrug())
+                g2.setPaint(DefaultRenderConstants.DEFAULT_DRUG_FOREGROUND);
+            else
+                g2.setPaint(DefaultRenderConstants.DEFAULT_FOREGROUND);
```

Comment:
I believe this is wrong without using parentheses. Also how about node.getIsForDisease()?